docs/user/01-20-istio-custom-resource.md

```diff
@@ -0,0 +1,336 @@
+The `istios.operator.kyma-project.io` CustomResourceDefinition (CRD) describes the kind and the format of data that Istio Controller uses to configure, update, and manage the Istio installation. Applying the CR triggers the installation of Istio, and deleting it triggers the uninstallation of Istio. To get the up-to-date CRD in the `yaml` format, run the following command:
+
+```shell
+kubectl get crd istios.operator.kyma-project.io -o yaml
+```
+
+## Specification
+
+This table lists all the possible parameters of the given resource together with their descriptions:
+
+<!-- TABLE-START -->
+### Istio.operator.kyma-project.io/v1alpha2
+
+**Spec:**
```

Comment:
I think we need to limit the config to link to types that we use from Istio.
An abstraction on this level might be enough:

- cni.k8s.affinity - link to https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Affinity  and drop nested fields
- components.​cni.​k8s.​resources - link to https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Resources and drop nested fields
- components.​ingressGateway.​k8s.​resources - link to https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Resources and drop nested fields
- components.​ingressGateway.​k8s.​strategy - link to https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#DeploymentStrategy and drop nested fields
- components.​pilot.​k8s.​resources - link to https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Resources and drop nested fields
- components.​pilot.​k8s.​strategy - link to https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#DeploymentStrategy and drop nested fields
- components.​proxy.​k8s.​resources - link to https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#Resources and drop nested fields